typegate/src/config/types.ts

```diff
@@ -88,12 +85,48 @@ export type TypegateConfigBase = z.infer<typeof typegateConfigBaseSchema>;
 // These config entries are only accessible on a Typegate instance.
 // They are not read from a global variable to enable test isolation and configurability.
 export const syncConfigSchema = z.object({
-  redis_url: z.string().transform((s) => new URL(s)),
-  s3_host: z.string().transform((s) => new URL(s)),
-  s3_region: z.string(),
-  s3_bucket: z.string(),
-  s3_access_key: z.string(),
-  s3_secret_key: z.string(),
+  redis_url: z.string().transform((s, ctx) => {
+    if (s === undefined) {
+      return addMissingEnvVarIssue("TG_ADMIN_PASSWORD", ctx);
+    }
+
+    return new URL(s);
+  }),
```

Comment:
They all do the same thing, refactor into a single function maybe?

Reply:
I guess expect for the `redis_url` and `tg_secret`, makes sense 👍